pkg/client/typed_client.go

```diff
@@ -168,8 +167,32 @@ func (c *typedClient) List(ctx context.Context, obj ObjectList, opts ...ListOpti
 		Into(obj)
 }
 
-// UpdateStatus used by StatusWriter to write status.
-func (c *typedClient) UpdateStatus(ctx context.Context, obj Object, opts ...UpdateOption) error {
+func (c *typedClient) CreateSubResource(ctx context.Context, obj Object, subResource Object, subResourceName string, opts ...SubResourceCreateOption) error {
```

Comment:
`subResource` is an `Object` in `CreateSubResource` but a `string` in `UpdateSubResource` and `PatchSubResource`.
How about name it `subResourceObj Object` so that `subResource` can all be the string of name?